components/compositions/compositions-parser.service.js

```diff
@@ -258,7 +259,15 @@ export default function (
       }
       const layers = me.jsonToLayers(obj);
       layers.forEach((lyr) => {
-        HsMapService.addLayer(lyr, true);
+        $timeout(() => {
```

Comment:
This doesnt seem right. At least the timeout should be outside forEach, se we don't trigger as many digests (at 10 at short interval they fail and break application) as there are layers.

Reply:
Oh I havent realized that, tested on some 6 layers and it went all well. 

Timeout might not be needed tho...there was a bug on Agrihub in Firefox only once more. Sometimes app.js layer failed to load and it left a whole layermanager empty. Problem was I could replicate much and also taht when putting a breakpoint to the error location it just never bugged. So I tought little  timeout might be enough. but wanted to add try/catch to make sure...but without timeout it would still be almost the same